### PR TITLE
Added framework matrix to build

### PR DESCRIPTION
**Pull request by josephmyers**
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/1362)

### CODE REVIEW

**Comment by josephmyers** [comment]
@hahn-kev also note the mildly terrifying build failure here:
https://github.com/sillsdev/libpalaso/actions/runs/11739761597/job/32704918933?pr=1362

Are we running into another concurrency issue with this design?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -21,11 +32,17 @@ jobs:
...
+
+    - name: Print
+      run: echo "$TEST_PATH"
+
     # there are cases where this will fail and we want to know about it
     # so we don't use continue-on-error, but we still want to publish the results
     - name: Test project
       id: test
-      run: dotnet test -bl:test.binlog --filter "TestCategory != SkipOnTeamCity" --blame-hang-timeout 5m --logger:"trx;LogFilePrefix=results" --results-directory ./test-results
+      run: dotnet test "$TEST_PATH"/SIL*Tests.dll --filter "TestCategory != SkipOnTeamCity" --blame-hang-timeout 5m --logger:"trx;LogFilePrefix=results" --results-directory ./test-results
```
Comment (hahn-kev) [inline .github/workflows/build.yml:45]: Instead of `$TEST_PATH` you want `${{ env.TEST_PATH }}` the same syntax as the concurrency value above

Comment (josephmyers) [reply .github/workflows/build.yml:45]: Interesting. Why is that? That variable appears to be working

**Comment by josephmyers** [comment]
> @hahn-kev also note the mildly terrifying build failure here: https://github.com/sillsdev/libpalaso/actions/runs/11739761597/job/32704918933?pr=1362
> 
> Are we running into another concurrency issue with this design?

We are. The build itself is flaky, where this work should fix the test flakiness. The build flakiness is hopefully much rarer

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -43,4 +55,22 @@ jobs:
         name: binary-logs
```
Comment (hahn-kev) [inline .github/workflows/build.yml:55]: This guy needs a unique name

**Review by hahn-kev — CHANGES_REQUESTED** [review]
Looks good! Missed the bin log upload name though

**Review by hahn-kev — APPROVED** [review]

**Comment by josephmyers** [comment]
Note that the Appveyor build didn't fail, it wasn't canceled due to my merging

**Pull request merged**